util/sync_buckets.py

```diff
@@ -0,0 +1,252 @@
+                    file,
+                    dest_bucket,
+                )
+                continue
+
+            md5 = _checksum(file_path, base64_encoded=True, algorithm=HashingAlgorithm.MD5)
+            logging.info("Computed md5 checksum for S3 upload: %s", md5)
+            with open(file_path, "rb") as data:
+                doc_manager.upload(dest_bucket, file, data, dryrun=not args.deploy, md5=md5)
+
+
+def _check_file_integrity(file, checksum_file, algorithm):
+    logging.info("Validating checksum for file %s", file)
+    with open(checksum_file, "r") as f:
+        expected_checksum = f.read().split(" ")[0]
+    file_checksum = _checksum(file, False, algorithm)
+    if expected_checksum != file_checksum:
+        raise Exception("Computed checksum %s does not match expected one %s", file_checksum, expected_checksum)
+
+
+def _download_file(url, file_path):
+    logging.info("Downloading file %s and saving it to %s", url, file_path)
+    urllib.request.urlretrieve(url, file_path)
+    logging.info("Validating size of downloaded file")
+    metadata = _get_s3_object_metadata(url)
+    downloaded_file_size = os.stat(file_path).st_size
+    if downloaded_file_size != int(metadata["size"]):
+        raise Exception(
+            f"Size of S3 object ({metadata['size']}) does not match size of downloaded file "
+            f"({downloaded_file_size})"
+        )
+
+
+def _download_files(args, dir):
+    bucket_url = (
+        f"https://{args.src_bucket}.s3.{args.src_bucket_region}.amazonaws.com"
+        f"{'.cn' if args.src_bucket_region.startswith('cn-') else ''}"
+    )
+    for file in args.src_files:
+        file_path = f"{dir}/{file}"
+        os.makedirs(os.path.dirname(file_path), exist_ok=True)
+        url = f"{bucket_url}/{file}"
+        _download_file(url, file_path)
+        if args.integrity_check:
+            checksum_file = f"{file_path}.{args.integrity_check}"
+            _download_file(f"{url}.{args.integrity_check}", checksum_file)
+            _check_file_integrity(file_path, checksum_file, args.integrity_check)
+
+
+def _validate_uploaded_files(args, uploaded_files, rollback_data):
+    for region in args.regions:
+        bucket_name = f"{args.dest_bucket.format(region=region)}"
+        bucket_url = f"https://{bucket_name}.s3.{region}.amazonaws.com{'.cn' if region.startswith('cn-') else ''}"
+        for file in uploaded_files:
+            url = f"{bucket_url}/{file}"
+            logging.info("Validating file %s", url)
+            metadata = _get_s3_object_metadata(url)
+            if not metadata["version_id"]:
+                logging.error("Cannot fetch object version")
+            if metadata["version_id"] == rollback_data[bucket_name]["files"][file]:
+                logging.error(f"Current version {metadata['version_id']} is the same as previous one")
+
+
+def main():
+    args = _parse_args()
+    logging.info("Parsed cli args: %s", vars(args))
+
+    checksum_files = []
```

Comment:
Could you please add a comment to explain that checksum_files should already exist in the source bucket next to src_files?

Reply:
it's documented as part of the "--integrity-check" option